geo/src/algorithm/sweep/line_or_point.rs

```diff
@@ -126,45 +143,54 @@ impl<T: GeoNum> PartialEq for LineOrPoint<T> {
 /// centered at its coordinates.
 impl<T: GeoNum> PartialOrd for LineOrPoint<T> {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        match (self.is_line(), other.is_line()) {
-            (false, false) => {
-                if self.left == other.left {
+        match (self, other) {
+            (LineOrPoint::Point(p), LineOrPoint::Point(o)) => {
+                if p == o {
                     Some(Ordering::Equal)
                 } else {
                     // Unequal points do not satisfy pre-condition and
                     // can't be ordered.
                     None
                 }
             }
-            (false, true) => other.partial_cmp(self).map(Ordering::reverse),
-            (true, false) => {
-                let (p, q) = self.end_points();
-                let r = other.left;
-                if r > q || p > r {
+            (LineOrPoint::Point(_), LineOrPoint::Line { .. }) => {
+                other.partial_cmp(self).map(Ordering::reverse)
+            }
+            (LineOrPoint::Line { left, right }, LineOrPoint::Point(p)) => {
+                if p > right || left > p {
                     return None;
                 }
                 Some(
-                    T::Ker::orient2d(*p, *q, *r)
+                    T::Ker::orient2d(**left, **right, **p)
                         .as_ordering()
                         .then(Ordering::Greater),
                 )
             }
-            (true, true) => {
-                let (p1, q1) = self.end_points();
-                let (p2, q2) = other.end_points();
-                if p1 > p2 {
+            (
+                LineOrPoint::Line {
+                    left: left_a,
+                    right: right_a,
+                },
+                LineOrPoint::Line {
+                    left: left_b,
+                    right: right_b,
+                },
+            ) => {
+                if left_a > left_b {
                     return other.partial_cmp(self).map(Ordering::reverse);
                 }
-                if p1 >= q2 || p2 >= q1 {
+                if left_a >= right_b || left_b >= right_a {
                     return None;
                 }
 
                 // Assertion: p1 <= p2
                 // Assertion: pi < q_j
                 Some(
-                    T::Ker::orient2d(*p1, *q1, *p2)
+                    T::Ker::orient2d(**left_a, **right_a, **left_b)
                         .as_ordering()
-                        .then_with(|| T::Ker::orient2d(*p1, *q1, *q2).as_ordering()),
+                        .then_with(|| {
+                            T::Ker::orient2d(**left_a, **right_a, **right_b).as_ordering()
+                        }),
```

Comment:
Here are my changes, the error isn't even showing the code from the previous version 👀 I `rg`-d over the whole repo and I couldn't find the code that was mentioned in the error message ...